computervision/src/main/java/ComputerVisionInAction.kt

```diff
@@ -16,23 +17,28 @@ internal class ComputerVisionInAction(private val cvPortal: SensorPortal) : Comp
     }
 
     private fun cvTokenize(rawData: String): Vision? {
-        val raw = JSONObject(rawData)
-        return when (raw.getInt("type") == 1) {
-
-            raw.has("value") -> Vision.OCR(raw.optString("value"))
-            raw.has("value") -> Vision.Scenery(raw.optString("value"))
-            raw.has("value") -> Vision.Facial(raw.optString("value"))
-            raw.has("value") -> Vision.Emotion(raw.optString("value"))
-            raw.has("value") -> Vision.ObjectDetection(raw.optJSONArray("value"))
-            else -> null // (corrupt reading, discard.) the russians did it for cv !
+        try {
+            val raw = JSONObject(rawData)
+            val list: List<Any> = listOf<Any>(raw.getJSONArray("value")) //[[......]]
+
+            return when (raw.getInt("type")) {
+                OCR -> Vision.OCR(raw.getString("value"))
+                scenes -> Vision.Scenery(raw.getString("value"))
+                prettyFaces -> Vision.Facial(raw.getString("value"))
+                emotions -> Vision.Emotion(raw.getString("value"))
+                objects -> Vision.ObjectDetection(list)
+                else -> null // (corrupt reading, discard.) the russians did it for cv !
+            }
+        }catch(e: JSONException){
+            return null
         }
     }
 
     private fun cvEncode(processed: CVInput): String {
-
         return when (processed) {
             is CVInput.ModeChange -> {
-                "${processed.mode.optInt("value")}"
+                    "${processed.mode}"
```

Comment:
don't forget, `type_value` in here:
```kotlin
"0_${processed.mode}"
```
where `0` means a `MODE_CHANGE` for instance. Also, `0_$` makes an interesting face.